rasterio/session.py

```diff
@@ -112,6 +112,9 @@ def cls_from_path(path):
         elif path.path.startswith("/vsiswift/"):
             return SwiftSession
 
+        elif path.scheme == "az":
```

Comment:
```suggestion
        elif path.path.startwith("/vsiaz/"):
```

What would you think about this for 1.1.4 @drnextgis? I don't it's the right time to pick a scheme for Azure. The Azure team, at least, should get some input in this.

Reply:
In my case I have database with links to rasters stored either in S3 ("s3://") or in Azure Blob Storage ("azblob://"). The intent of this PR was adding possibility to open these rasters directly with rasterio without any preprocessing of URLs. If we will use "/vsiaz" as you suggested then in my case it means I will have to rename URLs in database from "azblob://" to GDAL-specific URLs. This is what I wouldn't like to do. By the way I can open /vsiaz URLs now without any code changes:

```
with rasterio.Env(
    AZURE_STORAGE_ACCOUNT="...",
    AZURE_STORAGE_ACCESS_KEY="...",
):
...
```
@sgillies could you please explain what will be the value of this PR if we will go the way you have suggested?